jaeger-client/build.gradle

```diff
@@ -7,3 +9,28 @@ dependencies {
 
     testCompile group: 'junit', name: 'junit', version: junitVersion
 }
+
+/* Shaded the same libraries as jaeger-thrift,
+ * leaving the slf4j dependency in place, as a *requirement*
+ * for a standalone fat jar. */
+shadowJar {
+    relocate 'com.google.gson'  , 'io.jaegertracing.vendor.com.google.gson'
+    relocate 'com.twitter'      , 'io.jaegertracing.vendor.com.twitter'
+    relocate 'okhttp'           , 'io.jaegertracing.vendor.okhttp'
+    relocate 'okio'             , 'io.jaegertracing.vendor.okio'
+    relocate 'org.apache'       , 'io.jaegertracing.vendor.org.apache'
+    relocate 'org.slf4j'        , 'io.jaegertracing.vendor.org.slf4j'
+    classifier 'bundle'
+    dependencies {
+        exclude(dependency('org.projectlombok:lombok'))
+        exclude(dependency('org.codehaus.mojo:animal-sniffer-annotations'))
+    }
+}
+
+artifacts {
+    archives(shadowJar.archivePath) {
+        builtBy shadowJar
+    }
+}
+
+tasks.check.dependsOn tasks.shadowJar
```

Comment:
This wasn't required in the thrift usecase - was the shadowJar task not being triggered?

Reply:
`shadowJar` is *only* being triggered in the thrift case when `jaeger-crossdock` is being built. Based on what you wrote, it *sounds* like we should do the same here? :)

Reply:
Not sure it should be dependent on `jaeger-crossdock` build - ideally it should just be built when the non-shaded dependency is being built.

Reply:
Oh, then this line should stay, as it builds the shaded artifact right after the non-shaded one was built.